include/sentry.h

```diff
@@ -1313,28 +1328,39 @@ SENTRY_EXPERIMENTAL_API void sentry_transaction_context_set_sampled(
  * Transaction constructed off of the context.
  *
  * The Transaction will use the sampling rate as defined in `sentry_options`.
+ *
+ * The Transaction Context passed in must have an exclusive write lock to be
+ * thread-safe.
  */
 SENTRY_EXPERIMENTAL_API void sentry_transaction_context_remove_sampled(
     sentry_transaction_context_t *tx_cxt);
 
 /**
  * Starts a new Transaction based on the provided context, restored from an
  * external integration (i.e. a span from a different SDK) or manually
- * constructed by a user. Returns a Transaction, which is expected to be
- * manually managed by the caller. Manual management involves ensuring that
- * `sentry_transaction_finish` is invoked for the Transaction, and that the
- * caller manually starts and finishes any child Spans as needed on the
- * Transaction.
+ * constructed by a user.
+ *
+ * Returns a Transaction, which is expected to be manually managed by the
+ * caller. Manual management involves ensuring that `sentry_transaction_finish`
+ * is invoked for the Transaction, and that the caller manually starts and
+ * finishes any child Spans as needed on the Transaction.
  *
- * `sentry_transaction_finish` must be called in order for this Transaction to
- * be sent to sentry.
+ * Not invoking `sentry_transaction_finish` with the returned Transaction means
+ * it will be discarded, and will not be sent to sentry.
  *
  * To ensure that any Events or Message Events are associated with this
  * Transaction while it is active, invoke and pass in the Transaction returned
  * by this function to `sentry_set_span`. Further documentation on this can be
  * found in `sentry_set_span`'s docstring.
  *
- * Takes ownership of `transaction_context`.
+ * Takes ownership of `transaction_context`. The Transaction Context passed in
+ * must have an exclusive write lock to be thread-safe.
```

Comment:
```suggestion
 * Takes ownership of `transaction_context` and this should no longer be mutated by user code.
```

I'm not sure this second sentence still makes sense here.  "Taking ownership" means you need to never use the thing again.